strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java

```diff
@@ -200,14 +202,177 @@ public void download(@ApiParam(value = "The storageId", required = true)
             return;
         }
 
-        setMediaTypeHeader(path, response);
-
-        response.setHeader("Accept-Ranges", "bytes");
+        RepositoryFileAttributes fileAttributes;
+        try
+        {
+            fileAttributes =  getArtifactManagementService().getAttributes(storageId, repositoryId, path);
+            
+            if (fileAttributes == null)
+            {
+                response.setStatus(NOT_FOUND.value());
+                return;
+            }
+        }
+        catch (ArtifactTransportException e)
+        {
+            logger.debug("Unable to find artifact by path " + path, e);
 
-        ArtifactControllerHelper.setHeadersForChecksums(is, response);
+            response.setStatus(NOT_FOUND.value());
 
+            return;
+        }
+        logger.debug("RETURNED ATTRIBUTES");
+        setHeaders(is, response, fileAttributes, path);
+        
         logger.debug("Download succeeded.");
     }
+    
+    @ApiOperation(value = "Used to retrieve headers of an artifact", position = 2)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = ""),
+                            @ApiResponse(code = 400, message = "An error occurred.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(value = { "{storageId}/{repositoryId}/{path:.+}" }, method = RequestMethod.HEAD)
+    public void getHeaders(@ApiParam(value = "The storageId", required = true)
+                           @PathVariable String storageId,
+                           @ApiParam(value = "The repositoryId", required = true)
+                           @PathVariable String repositoryId,
+                           @RequestHeader HttpHeaders httpHeaders,
+                           @PathVariable String path,
+                           HttpServletRequest request,
+                           HttpServletResponse response)
+            throws Exception
+    {
+        logger.debug("Requested Headers for /" + storageId + "/" + repositoryId + "/" + path + ".");
+
+        Storage storage = configurationManager.getConfiguration().getStorage(storageId);
+        if (storage == null)
+        {
+            logger.error("Unable to find storage by ID " + storageId);
+
+            response.sendError(INTERNAL_SERVER_ERROR.value(), "Unable to find storage by ID " + storageId);
+
+            return;
+        }
+
+        Repository repository = storage.getRepository(repositoryId);
+        if (repository == null)
+        {
+            logger.error("Unable to find repository by ID " + repositoryId + " for storage " + storageId);
+
+            response.sendError(INTERNAL_SERVER_ERROR.value(),
```

Comment:
Same as above:

This should be  a `404` error.
You should not be returning error `500`, as it's more for `RuntimeException`-s and the likes.